app/views.py

```diff
@@ -8,84 +8,146 @@
+
     cards = CardsRepository.get_by_deck_id(deck_id)
 
+    log.info("Cards records are retrieved from database")
+    log.info("Rendering show_cards.html")
+
     return render_template("show_cards.html", deck_id=deck_id, cards=cards)
 
 
 @app.route("/deck/<deck_id>/card/<card_id>")
 def card_detail(deck_id, card_id):
+
+    log.info("Route '/deck/<deck_id>/card/<card_id>' is called")
+
     card = CardsRepository.get_by_id(card_id)
 
+    log.info("Card is retrieved from database")
+    log.info("Rendering card_detail.html")
     return render_template("card_detail.html", card=card)
 
 
 @app.route("/create_deck", methods=["POST", "GET"])
 def create_deck():
+
+    log.info("Route '/creatE_deck' is called")
+
     if 'user' not in session:
+        log.info("User is not authenticated")
+        log.info("Redirecting to route '/login'")
         return redirect(url_for("login"))
 
+    log.info('User is authenticated')
+
     if request.method == "POST":
 
+        log.info("Form is submitted")
+
         name = request.form['name']
 
         user_id = json.loads(session['user'])['id']
 
+        log.info("Form is processed")
+
         deck = DecksRepository.create(user_id, name)
 
         if deck:
+            log.info("Deck is created and saved into database")
+            log.info("Rendering create_deck.html")
             return render_template("create_deck.html", success=True)
         else:
+            log.info("Deck is not created and is not saved into database")
+            log.info("Rendering create_deck.html")
             return render_template("create_deck.html", success=False)
 
     else:
+        log.info('Rendering create_deck.html')
         return render_template("create_deck.html")
 
 
-@app.route("/deck/<id>/create_card", methods=["GET", "POST"])
-def create_card(card_id):
+@app.route("/deck/<deck_id>/create_card", methods=["GET", "POST"])
+def create_card(deck_id):
+    log.info("Route '/deck/<deck_id>/create_card' is called")
+    
     if 'user' in session:
-
+        log.info("User is authenticated")
         if request.method == "POST":
-
+            
+            log.info("Form is submitted")
             word = request.form['word']
             translation = request.form["translation"]
 
-            if CardsRepository.create(card_id, word, translation):
+            if CardsRepository.create(deck_id, word, translation):
+                log.info("Card is created and saved into database")
+                log.info("Rendering response")
                 return "<h2>New card has been created</h2>"
-
+            
+            log.info("Card is not created")
+            log.info("Rendering response")
             return "<h2>There was some error</h2>"
 
         else:
+
+            log.info("Rendering create_card.html")
             return render_template("create_card.html")
 
+    log.info("User is not authenticated")
+    log.info("Redirecting to route '/login'")
     return redirect(url_for("login"))
```

Comment:
maybe if user is not authenticated then redirect to login, and then write whats after log.info("User is authenticated")
but you dont need else then

Reply:
don't understand

Reply:
if 'user' not in session:
return redirect(url_for("login"))

then you  can write rest of the method without writing else

Reply:
corrected, done?